src/query/expression/src/values.rs

```diff
@@ -1409,55 +1441,46 @@ impl Column {
                         .values()
                         .clone(),
                 ),
-                ArrowDataType::LargeBinary => {
-                    let arrow_col = arrow_col
-                        .as_any()
-                        .downcast_ref::<databend_common_arrow::arrow::array::BinaryArray<i64>>()
-                        .expect("fail to read from arrow: array should be `BinaryArray<i64>`");
-                    let offsets = arrow_col.offsets().clone().into_inner();
-
-                    let offsets =
-                        unsafe { std::mem::transmute::<Buffer<i64>, Buffer<u64>>(offsets) };
-                    // LargeBinary may be Extension data type variant and bitmap
```

Comment:
It is recommended to keep this code logic, I forgot exactly what scenarios, but they should be useful.

Reply:
Let me test it.